DatadogRUM/Tests/Instrumentation/Resources/URLSessionRUMResourcesHandlerTests.swift

```diff
@@ -940,6 +940,74 @@ class URLSessionRUMResourcesHandlerTests: XCTestCase {
         XCTAssertTrue(baggageHeader?.contains("session.id=abcdef01-2345-6789-abcd-ef0123456789") == true)
     }
 
+    // MARK: - GraphQL Header Extraction Tests
+
+    func testGivenRequestWithGraphQLHeaders_whenInterceptionCompletes_itExtractsGraphQLAttributes() throws {
+        let receiveCommand = expectation(description: "Receive RUMStopResourceCommand")
+        var stopResourceCommand: RUMStopResourceCommand?
+        commandSubscriber.onCommandReceived = { command in
+            if let command = command as? RUMStopResourceCommand {
+                stopResourceCommand = command
+                receiveCommand.fulfill()
+            }
+        }
+
+        // Given
+        var mockRequest = URLRequest(url: URL(string: "https://graphql.example.com/api")!)
```

Comment:
We have [a mock](https://github.com/DataDog/dd-sdk-ios/blob/c1b003feaa2dc992b6e83a9c95aa5c1099559678/TestUtilities/Sources/Mocks/SystemFrameworks/FoundationMocks.swift#L566C24-L566C32) you can use for this:
```suggestion
        var mockRequest: URLRequest = .mockWith(url: "https://graphql.example.com/api")
```